AppCenterData/AppCenterData/Internal/LocalStore/MSDocumentStore.h

```diff
@@ -87,6 +88,56 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (MSDocumentWrapper *)readWithToken:(MSTokenResult *)token documentId:(NSString *)documentId documentType:(Class)documentType;
 
+/**
+ * List all the documents from the store and return the list if not expired.
+ *
+ * @param token CosmosDB token.
+ * @param partition The CosmosDB partition key.
+ * @param documentType The document type to list.
+ * @param baseOptions Options for listing and storing the documents.
+ *
+ * @return A MSPaginatedDocuments object. List of documents found in the local store for a particular partition.
+ */
+- (MSPaginatedDocuments *)listWithToken:(MSTokenResult *)token
+                              partition:(NSString *)partition
+                           documentType:(Class)documentType
+                            baseOptions:(MSBaseOptions *_Nullable)baseOptions;
+
+/**
+ * Checks if there are any pending operations to be processed.
+ *
+ * @param partition The CosmosDB partition key.
+ *
+ * @return true if the partition has pending operations, else returns false.
+ */
+- (BOOL)hasPendingOperationsForPartition:(NSString *)partition;
+
+/**
+ * Update the local store given a current/new cached document.
+ *
+ * @param token The CosmosDB token.
+ * @param currentCachedDocument The current cached document.
+ * @param newCachedDocument The new document that should be cached.
+ * @param deviceTimeToLive The device time to live for the new cached document.
+ * @param operation The operation being intended (nil - read, CREATE, UPDATE, DELETE).
```

Comment:
Why READ operation is not a contant, too? It would be more explicit. Not blocking, just my opinion :)